cips/cip-blob-verified-signer.md

```diff
@@ -0,0 +1,92 @@
+message Blob {
+  bytes namespace_id = 1;
+  bytes data = 2;
+  uint32 share_version = 3;
+  uint32 namespace_version = 4;
+  // new field
+  string signer = 5;
+}
+```
+
+Given proto's backwards compatibility, users could still submit the old blob type (in the `BlobTx` format) and signer would be processed as an empty string.
+
+The new block validity rule (In `PrepareProposal` and `ProcessProposal`) would thus be that if the signer was not empty, then it must match that of the PFB that paid for it. When validating the `BlobTx`, validators would check the equivalency of the PFB's `signer` to the Blob's `signer` (as well as verification of the signature itself).
+
+Although no version changes are required for protobuf encoded blobs, share encoding would change. Blobs containing a non empty signer string would be encoded using the new v2 format:
+
+![Diagram of V2 Share Format](../assets/cip-blob-verified-signer/blob-v2-share-format.svg)
+
+Blobs with an empty `signer` string would remain encoded using the v1 format. Note that in this diagram it is the `Info Byte` that contains the share version. Not to be confused with the namespace version.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+## Rationale
+
+Given the current specification change, the new loop is simplified:
+
+- Retrieve all blobs in the subscribed namespaces
+- Verify the namespaces inclusion
+- Verify that the `signer` in each blob matches that of an allowed sequencer
+
+As a small digression, it may be feasible to additionally introduce a new namespace version with the enforcement that all blobs in that namespace use the v2 format i.e. have a signer. However, this does not mean that the signer matches that of the sequencer (which Celestia validators would not be aware of). This would mean that full nodes would need to get and verify all blobs in the namespace anyway.
+
+## Backwards Compatibility
+
+This change requires a hard fork network upgrade as older nodes will not be able to verify the new blob format. The old blob format will still be supported allowing rollups to opt into the change as they please.
+
+## Test Cases
+
+Test cases will need to ensure that a user may not forge a incorrect signer, nor misuse the versioning. Test cases should also ensure that the properties of backwards compatibility mentioned earlier are met.
+
+## Reference Implementation
+
+TBC
+
+## Security Considerations
+
+Rollups using this pattern for verifying the enshrined sequencer make an assumption that there is at least 1/3 in voting power of the network is "correct". Note this is a more secure assumption than forking which may require up to 2/3+ of the voting power to be "correct". Rollups may decide to still retrieve the PFB's and validate the signatures themselves if they wish to avoid this assumption.
```

Comment:
I don't follow this security consideration. 

> Note this is a more secure assumption than forking which may require up to 2/3+ of the voting power to be "correct".

Does "forking" in this context apply to Celestia or the roll-up? Why do 2/3 of the voting power need to be correct to fork Celestia? Hypothetically if 2/3 voting power on the Celestia consensus layer are malicious, the 1/3 could create a fork that excludes the 2/3 malicious nodes.

> Rollups may decide to still retrieve the PFB's and validate the signatures themselves if they wish to avoid this assumption.

I think this is trying to say that regardless of the number of "incorrect" voting power on the DA layer, the DA consensus layer can't include a PFB with an signature that would convince an honest roll-up node to be deceived. But I don't really understand how verifying the PFB signatures avoids an assumption from above about 1/3 or 2/3 correct voting power.

Reply:
> Does "forking" in this context apply to Celestia or the roll-up? 

To Celestia (which also implies that there is a potential fork of all the rollups on Celestia).

What I was saying is that there is an assumption on the honesty of the validator set when using the signer (since the validators validate that the signer is correct) but that this assumption is in any way outweighed by the bigger assumption (i.e. the less secure attack vector) that there isn't currently a fork on Celestia

Reply:
Ahh got it, thanks for expanding here.

Reply:
is this resolved?

Reply:
Yea resolved.